interview-questions.md

```diff
@@ -15,41 +15,52 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 
   Your answer:
+  An index is each "part" of an array. If an array has strings, than the indexes are each string that is separated by a comma. If an array has numbers, each number separated by commas are counted as an index. Indexes start from zero as the first one in the array and are held within brackets. The difference between index and value, is that a value is held within parentheses and is usually a parameter assigned to a function or method. An index is held within brackets as an array. You can return a value from an index.
 
   Researched answer:
+  The index is a location of the value within an array. The index serves as a place to find the given value you are seeking. It does, however, start in the array at zero. The array as a whole is known as an element and the contents of it is known as it's value. For example, [1, 2, 3]: 1 is the value but it is held at index 0.
+
 
 
 
 4. What is iteration?
 
   Your answer:
+  I am not too familiar with the term iteration at this time, but I will get back to you on the term when I have a better understanding of it.
 
   Researched answer:
+  Iteration is when a code can be ran again and again in order to meet a condition. The best way I can explain this, is by showing how for loops preform. For loops will take a given starting index, a given stop index (or how many times you want the iteration to preform), and a given number of intervals of indexes to act on. The iteration, if not given a specific place to stop or condition to be met, will contiue to run like the legendary Forrest Gump.
+
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
+  Proper pair programming techniques can look like two people bouncing off of each other with insight on how to tackle problems within code or even just developing code in general. There is amazing brain storming between the two minds working on the same output and how to achieve it in the best practical way. Pair programming techniques include a lot of team work, communication, and open mindness. If done effectively, pair programming can expand the knowledge and perspective of both developers. The roles include a "Driver" and a "Navigator". The "Driver" is the person with "hands on code" for instance. They are the one actually manipulating the code, while the "Navigator" is the one overseeing and giving input. Ideally the "Navigator" is giving direction, looking for grammar and syntax, and talking through the coding with the "Driver".
 
   Researched answer:
+  Techniques best practiced in pair programming is switching! There is something known as the pomodoro method, in which a task is decided upon and assigned to each person so that the pair can alternate different parts of the project with a set timer to alert each person that it is time to switch roles. Incorporating short breaks to battle fatigue and regain focus.
```

Comment:
Excellent answers in this section!  Every answer was well thought out, thorough and just what I would look for when interviewing someone.  Great job!